imports/ui/reducers/index.js

```diff
@@ -136,6 +136,16 @@ const songsReducer = (
             : song;
         })
       };
+    case 'ADD_TAG_FAILURE':
+      return {
+        ...state,
+        isInputDisabled: false
```

Comment:
isInputDisabled isn't part of state in songsReducer (Line 111 of this file). See what other change you can make to the state to handle ADD_TAG_FAILURE or REMOVE_TAG_FAILURE

Reply:
If there's no change in state for the song if there's a failure, I think we can remove these 2 actions altogether and just display the notification for error.